### PR TITLE
test VarWebScaleAndColormap with wrong input

### LINKED ISSUE #1291: Arguments given in  VarWebScaleAndColormap may not be going through pydantic
**Describe the bug**

There are two `__init__` methods in this class. Only the second one is being used. This means that the pydantic checking is not being used. One way to test this is to try and give `VarWebScaleAndColormap` arguments which are not `scale` and `colmap`

 - Pyaerocom version: 0.21.dev0
 - Computing platform: All
 - Configuration file (if applicable): NA
 - Error message (if applicable): NA

Relevant PR: #1256

### PR DESCRIPTION
**Pull request by heikoklein**
## Change Summary

testing VarWebScaleAndColormap to fail on wrong input

## Related issue number
fix #1291

## Checklist

* [ ] Start with a draft-PR
* [x] The PR title is a good summary of the changes
* [x] PR is set to AeroTools and a tentative milestone
* [x] Documentation reflects the changes where applicable
* [x] Tests for the changes exist where applicable
* [x] Tests pass locally
* [x] Tests pass on CI
* [x] At least 1 reviewer is selected
* [x] Make PR ready to review

### CODE REVIEW

File: pyaerocom/aeroval/glob_defaults.py
Comment (lewisblake) [inline pyaerocom/aeroval/glob_defaults.py:1]: I have to admit I find the use of helper classes in this case unnecessarily convoluted and confusing. I think we could accomplish something similar more efficiently making more obvious use of `BaseModel`. For example, here is some untested code which communicates the idea but uses 2/3 as many lines of code.


```python
class ScaleAndColmap(BaseModel):
    scale: list[float] = Field(...)
    colmap: str = Field(...)


class VarWebScaleAndColormap(BaseModel):
    scale_colmaps: dict[str, ScaleAndColmap] = Field(...)

    @classmethod
    def from_ini(cls, filename: str):
        cfg = ConfigParser()
        if not os.path.exists(filename):
            raise FileNotFoundError(
                f"{cls.__name__} from_ini('{filename}') which does not exist"
            )
        cfg.read(filename)
        # remove configparser default
        cfg_dict = dict()
        keys = ("scale", "colmap")
        for s in cfg.sections():
            items = []
            for k in keys:
                if k in cfg[s]:
                    if k == "scale":
                        try:
                            items.append(json.loads(cfg[s][k]))
                        except Exception as ex:
                            raise KeyError(
                                f"wrong value for '{k}' of var '{s}' in {filename}: {ex}"
                            )
                    else:
                        items.append(cfg[s][k])
                else:
                    raise KeyError(f"missing '{k}' for var '{s}' in {filename}")
            cfg_dict[s] = {"scale": items[0], "colmap": items[1]}
        return cls(scale_colmaps=cfg_dict)

    def update_from_ini(self, filename: str):
        updated_model = self.from_ini(filename)
        self.scale_colmaps.update(updated_model.scale_colmaps)

    def write(self, filename: str):
        """write the scales and colormaps to a configuration file

        :param filename: path to a filename, currently a .ini file
        """
        cfg = ConfigParser()
        cfg.read_dict(self.model_dump())
        with open(filename, "w") as fh:
            cfg.write(fh)
            
```

Comment (lewisblake) [reply pyaerocom/aeroval/glob_defaults.py:1]:
```python
# usage
vwsac = VarWebScaleAndColormap.from_ini("var_scale_colmap.ini")
vwsac.update_from_ini("other_file.ini")
vwsac.write("output.ini")
```

File: tests/aeroval/test_glob_defaults.py
Comment (lewisblake) [inline tests/aeroval/test_glob_defaults.py:1]: Thanks for adding these tests!

File: pyaerocom/aeroval/glob_defaults.py
Comment (heikoklein) [reply pyaerocom/aeroval/glob_defaults.py:1]: pydantic.BaseModel is a tool, not a solution. Not using BaseModel as an exposed part of the API is a clean design choice. We might switch to another validator at some time, which will be hard if a user relies on anything from BaseModel.

**Review by lewisblake — APPROVED** [review]

**Pull request merged**